syntax = "proto3";
package protos.playbooks;

import "google/protobuf/wrappers.proto";
import "google/protobuf/struct.proto";


message Api {
  message HttpRequest {
    enum Method {
      UNKNOWN = 0;
      GET = 1;
      POST = 2;
      PUT = 3;
      PATCH = 4;
      DELETE = 5;
    }
    Method method = 1;
    google.protobuf.StringValue url = 2;
    google.protobuf.StringValue headers = 3;
    google.protobuf.StringValue payload = 4;
    google.protobuf.UInt64Value timeout = 5;
    google.protobuf.StringValue cookies = 6;
  }

  enum TaskType {
    UNKNOWN = 0;
    HTTP_REQUEST = 1;
  }

  TaskType type = 1;
  oneof task {
    HttpRequest http_request = 2;
  }
}